### PR TITLE
EnumTypeModelBinder does not handle enum flags provided as a list

### LINKED ISSUE #10191: EnumTypeModelBinder does not handle enum flags correctly
### Describe the bug
When binding an flag enum type to checkbox lists, only the first checked box gets picked up.

### Expected behavior
When multiple checkboxes are checked, the bound model property should reflect selections.

### Additional context
The implementation for EnumTypeModelBinder derives from SimpleTypeModelBinder, which only looks at the first value from ValueProviderResult during BindModelAsync(). To fix this issue, line 65 in SimpleTypeModelBinder.cs should be changed from

`var value = valueProviderResult.FirstValue;`

to

`var value = valueProviderResult.ToString();`

### PR DESCRIPTION
**Pull request by silamon**
Fixes #10191. Flags enum can be provided as a comma separated list according to the existing test cases. However, when binded to a form coming from a razor page, these are provided an array list. This applies the same fix as previous tried, but also provide unit tests. 

If you checkout my other branch FlagsEnumProvidedAsListSample on my fork, you can find a working sample with a razor page in the MvcSandbox.

### CODE REVIEW

**Comment by captainsafia** [comment]
It looks like this changeset might've introduced test failures in existing tests. Can you take a look at this, @silamon?

**Comment by captainsafia** [comment]
I don't have much context into the original fix as it is about 3 years old now. Some of the commentary on the previous issue seems to indicate that there isn't much work needed in establishing the wire format.

Going back a few steps, what scenario does this bug fix address? 

cc: @brunolins16

**Comment by silamon** [comment]
If you bind a flag enum with razor pages to a checklist, only the first selected checkbox value is bound to the variable. This pull address fixes that so that all selected values are bound to the flag enum. 

While the tests now pass, I would prefer to move the enum logic out of simpletypemodelbinding.

**Comment by silamon** [comment]
You can find the documentation of the wire format in the codebase for context: https://github.com/dotnet/aspnetcore/blob/4d75ee9a7fbad112e29b8a1d77c7d96965f62b40/src/Mvc/Mvc.Core/src/ModelBinding/Binders/EnumTypeModelBinder.cs

**Comment by silamon** [comment]
@captainsafia have you been able to look into this already?

File: src/Mvc/Mvc.Core/test/ModelBinding/Binders/EnumTypeModelBinderTest.cs [hunk 1]
```diff
@@ -149,6 +149,91 @@ public async Task BindModel_AddsErrorToModelState_ForInvalidEnumValues(Type mode
         Assert.Equal(message, error.ErrorMessage);
     }
 
+    [Theory]
+    [InlineData("8, 1")]
+    [InlineData("Value2, Value8")]
+    [InlineData("value8,value4,value2,value1")]
+    public async Task BindModel_BindsTo_NonNullableFlagsEnumType_List(
+        string flagsEnumValue,
+        params string[] flagsEnumValueList
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/test/ModelBinding/Binders/EnumTypeModelBinderTest.cs:158]: Remove as these are unused?

**Review by captainsafia — COMMENTED** [review]
OK! I think I've done the due diligence here to be OK with this change despite the conversation that happened a few years ago. It looks like the wire format _is_ documented and the fact that the code changes only impact `[Flags]` enums makes me more comfortable trying this out.

I left one small note for cleaning up unused parameters on the test cases but other than that I think we should be good here.

Thank you for your patience! 🙏🏽

**Comment by silamon** [comment]
@dotnet-policy-service agree

**Review by captainsafia — APPROVED** [review]

**Pull request merged**